PollTimer.cpp

```diff
@@ -0,0 +1,42 @@
+#include "PollTimer.h"
+
+PollTimer::PollTimer(uint64_t logIntervalSeconds)
+    : _logPeriod(logIntervalSeconds * STIME_US_PER_S), _lastStart(0), _lastStop(0), _lastLogStart(0), _timeInPoll(0),
+      _timeNotInPoll(0) {}
+
+void PollTimer::startPoll() {
+    uint64_t timestamp = STimeNow();
+
+    if (_lastStop) {
+        _timeNotInPoll += timestamp - _lastStop;
+    }
+
+    _lastStart = timestamp;
+    if (!_lastLogStart) {
```

Comment:
Some comments here would be great to explain this logic.  I'm not really sure where you're trying to do here